File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {   
        int leftBound = 0;
        int rightBound = nums.length;
        
        while (leftBound != rightBound) {
            int curIdx = leftBound + (rightBound - leftBound) / 2;
            int curNum = nums[curIdx];
            if (curNum == target) {
                return curIdx;
            }
            if (curNum > target) {
                rightBound = curIdx;
            } else {
                leftBound = curIdx + 1;
            }
        }
        
        return -1;
    }
}

File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if not nums:
            return -1
        
        mid = len(nums) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] < target:
            recursiveResult = self.search(nums[mid+1:], target) + 1
            return mid + recursiveResult if recursiveResult > 0 else -1
        else:
            return self.search(nums[:mid], target)